widgets/heatmap/internal/axes/label.go

```diff
@@ -36,21 +38,85 @@ type Label struct {
 // the labels will be placed and aligned.
 // Labels are returned with Y coordinates in ascending order.
 // Y coordinates grow down.
-func yLabels(graphHeight, labelWidth int, labels []string) ([]*Label, error) {
-	return nil, errors.New("not implemented")
+func yLabels(labelWidth int, labels []string) ([]*Label, error) {
+	if min := 0; labelWidth < min {
+		return nil, fmt.Errorf("cannot place labels on a canvas with width %d, minimum is %d", labelWidth, min)
+	}
+
+	var ret []*Label
+	for row, l := range labels {
+		label, err := rowLabel(row, l, labelWidth)
+		if err != nil {
+			return nil, err
+		}
+		ret = append(ret, label)
+	}
+
+	return ret, nil
 }
 
-// rowLabel returns one label for the specified row.
+// rowLabel returns one Y label for the specified row.
 // The row is the Y coordinate of the row, Y coordinates grow down.
 func rowLabel(row int, label string, labelWidth int) (*Label, error) {
-	return nil, errors.New("not implemented")
+	// The area available for the Y label
+	ar := image.Rect(0, row, labelWidth, row+1)
+
+	pos, err := alignfor.Text(ar, label, align.HorizontalRight, align.VerticalMiddle)
+	if err != nil {
+		return nil, fmt.Errorf("unable to align the label value: %v", err)
+	}
+
+	return &Label{
+		Text: label,
+		Pos:  pos,
+	}, nil
 }
 
 // xLabels returns labels that should be placed under the cells.
 // Labels are returned with X coordinates in ascending order.
 // X coordinates grow right.
```

Comment:
Can we say something about the `yEnd` and `graphWidth` arguments to help any future readers?